streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/testing/EventCorrelationInjector.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2017 Hortonworks.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *   http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package com.hortonworks.streamline.streams.runtime.storm.testing;
+
+import com.google.common.collect.ImmutableMap;
+import com.hortonworks.streamline.streams.StreamlineEvent;
+import com.hortonworks.streamline.streams.common.StreamlineEventImpl;
+import com.hortonworks.streamline.streams.storm.common.StormTopologyUtil;
+import org.apache.storm.tuple.Tuple;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+// TODO: do we want to decouple this class with Storm specific?
+public class EventCorrelationInjector {
+
+    public static final String HEADER_KEY_ROOT_IDS = "rootIds";
+    public static final String HEADER_KEY_PARENT_IDS = "parentIds";
+    public static final String HEADER_KEY_SOURCE_COMPONENT_NAME = "sourceComponentName";
+
+    public StreamlineEvent injectCorrelationInformation(List<Object> tuple, List<Tuple> parentTuples, String componentName) {
```

Comment:
Maybe you can move this and any other methods that refer to Tuple out of this class and move the class to a common package that can be re-used else where.

Reply:
OK will do.